### PR TITLE
Feat pp file

### PR DESCRIPTION
**Pull request by jtwhite79**
Added support for predefined pilot points in pstfrom thru the pp_space arg.  Added some demo to the pstfrom mf6 notebook

### CODE REVIEW

File: pyemu/utils/pst_from.py [hunk 1]
```diff
@@ -1873,8 +2006,12 @@ def add_parameters(
                             "using ExpVario with contribution=1 "
                             "and a=(pp_space*max(delr,delc))"
                         )
-                        # set up a default
-                        pp_dist = pp_space * float(
+                        # set up a default - could probably do something better if pp locs are passed
```
Comment (briochh) [inline pyemu/utils/pst_from.py:2009]: I guess all fairly arbitrary, but could just use max separation of adjacent pp if pp_locs passed? But is that really any more "correct".

File: examples/pstfrom_mf6.ipynb [hunk 2]
```diff
@@ -361,11 +361,96 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "## Pilot points in `PstFrom`\n",
```
Comment (briochh) [inline examples/pstfrom_mf6.ipynb:364]: Nice. We should do more Notebooks.

**Review by briochh — APPROVED** [review]
Noice

File: pyemu/utils/pst_from.py [hunk 3]
```diff
@@ -1461,11 +1499,16 @@ def add_parameters(
                 pargp but is also used to gather correlated parameters set up
                 using multiple `add_parameters()` calls (e.g. temporal pars)
                 with common geostructs.
-            pp_space (`int`): Spacing between pilot point parameters
+            pp_space (`int`,`str` or `pd.DataFrame`): Spatial pilot point information.
+                If `int` it is the spacing in rows and cols of where to place pilot points.
+                If `pd.DataFrame`, then this arg is treated as a prefined set of pilot points
+                and in this case, the dataframe must have "name", "x", "y", and optionally "zone" columns.
+                If `str`, then an attempt is made to load a dataframe from a csv file (if `pp_space` ends with ".csv") or
+                from a pilot points file.  If `pp_space` is None, an integer spacing of 10 is used.  Default is None
```
[the remark this answers is not in the document]
Comment (jtwhite79) [reply pyemu/utils/pst_from.py:1507]: Ah yes - wrote the docstring before adding the shape file bit

File: pyemu/utils/pst_from.py (hunk 1 above)
Comment (jtwhite79) [reply pyemu/utils/pst_from.py:2009]: True - I'll add that bit

File: examples/pstfrom_mf6.ipynb (hunk 2 above)
Comment (jtwhite79) [reply examples/pstfrom_mf6.ipynb:364]: Yes!

**Pull request merged**